### DISCUSSION TITLE
Does Argo Rollout have a way to trigger HPA custom metrics?

### DISCUSSION THREAD

**Discussion by KenBarr** [argoproj/argo-rollouts#1521] [Q&A] [answered]
I would like to autoscale a rollout based on metrics other then CPU/Memory.  Specifically queue depth of upstream resource.  Does Argo Rollout have an interface to metrics.k8s.io, custom.metrics.k8s.io or external.metrics.k8s.io so that we can easily inject this kind of information into the HPA logic?  Am I missing something and there is a different way to do this via the Argo tool-chain?

**Comment by jessesuen** [comment, marked as answer]
This is not a question specific to Argo Rollouts. You will want to use the HPA custom metrics feature:
https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#support-for-custom-metrics

Rollouts should be compatible with that, as it applies to any HPA scalable workloads, including Rollouts.
